### ISSUE TITLE
primary attribute not taking effect in nested series

### ISSUE THREAD

**Issue by juliohm** [JuliaPlots/Plots.jl#1495]
Could you please confirm that the following should produce two series with the same color?

```julia
@userplot FooPlot

@recipe function f(foo::FooPlot)
  y = foo.args[1]
  x = 1:length(y)

  @series begin
    seriestype := :path
    primary := false

    x, y
  end

  seriestype --> :scatter
  
  x, 2y
end

fooplot(rand(3))
```
![screenshot from 2018-04-23 14-53-28](https://user-images.githubusercontent.com/3345261/39155264-230b2d18-4706-11e8-880b-ceadbf7f57f8.png)

I remember using it in the past, but I am not sure.

**Comment by joshday** [comment]
I believe `primary=false` is used for removing a series from the legend, which is what is happening in your example.  Questions like this may be better suited for slack or discourse.

**Comment by juliohm** [comment]
Thank you @joshday , I think this one is a real issue. I used to make use of the `primary` attribute for exactly that: maintain the previous style from the last plot command. It is not only about the legend. Check the example below, which still works as of today:

```julia
using Plots

x = linspace(0,1,10)

scatter(x)
plot!(x, primary=false)
```
![screenshot from 2018-04-24 07-37-52](https://user-images.githubusercontent.com/3345261/39194380-7174557e-4792-11e8-846f-75ca1d6fe697.png)

The color is preserved with built-in plot commands. The issue is in defining nested series in plot recipes. If there is any other way to circumvent this, I'd be happy to learn.

**Comment by mkborregaard** [comment]
You're right @juliohm . I think this was possibly introduced by https://github.com/JuliaPlots/Plots.jl/pull/1011 , but it's certainly an unwelcome side effect. Could you use this workaround?
```julia
@recipe function f(foo::FooPlot)
         y = foo.args[1]
         x = 1:length(y)

       @series begin
         seriestype --> :scatter
         x, 2y
       end


       @series begin
           seriestype := :path
           primary := false
           x, y
        end

         [],[]
       end
```

**Comment by juliohm** [comment]
Thank you @mkborregaard, that is a very nice solution.

**Comment by juliohm** [comment]
@mkborregaard the solution works nicely, except that the legend is affected. Even though the second series copies the color and style, the legend shows a different series (for the [], [] pair?) in the final plot.

**Comment by mkborregaard** [comment]
Yes it seems to do that in some backends but not others. Just adding `primary := false` to the outermost series (`[],[]') should solve this.

**Comment by matthieugomez** [comment]
Still, it sounds like this should be solved. For instance, seriestype= :scatterpath now gives different colors for the dots and the lines, which was not the intent.

**Comment by mkborregaard** [comment]
maybe just a question of fixing the scatterpath recipe?

**Comment by matthieugomez** [comment]
Yeah, maybe. It would give a good example if what to do exactly. In the
meantime, could this issue be reopened?

**Comment by matthieugomez** [comment]
Can be closed now #3423
